pls, fix compliance checks:
```
 Commit be5ad31e61:
1: UC6 Commit message body is empty, should at least have 1 line(s).

Commit a95cfc1473:
1: UC6 Commit message body is empty, should at least have 1 line(s).

Commit 73297c6b84:
1: UC6 Commit message body is empty, should at least have 1 line(s).
Error: Process completed with exit code 1.
```

--- a/drivers/tee/optee/optee.c
+++ b/drivers/tee/optee/optee.c
@@ -30,14 +30,6 @@ LOG_MODULE_REGISTER(optee);
  */
 #define TEE_OPTEE_CAP_TZ  BIT(0)
 
-/*
- * Bitmap of the ongoing notificatons, received from OP-TEE. Maximum number is
- * CONFIG_OPTEE_MAX_NOTIF. This bitmap is needed to handle case when SEND command
- * was received before WAIT command from OP-TEE. In this case WAIT will not create
- * locks.
- */
-SYS_BITARRAY_DEFINE_STATIC(notif_bitmap, CONFIG_OPTEE_MAX_NOTIF);
-
 struct optee_rpc_param {
 	uint32_t a0;
 	uint32_t a1;
@@ -87,6 +79,8 @@ struct optee_supp {
 
 struct optee_driver_data {
 	smc_call_t smc_call;
+
+	sys_bitarray_t *notif_bitmap;
 
 	sys_dlist_t notif;
 	struct k_spinlock notif_lock;
@@ -441,7 +435,7 @@ static int optee_notif_send(const struct device *dev, uint32_t key)
 	sp_key = k_spin_lock(&data->notif_lock);
 	if (!key_is_pending(data, key)) {
 		/* If nobody is waiting for key - set bit in the bitmap */
-		sys_bitarray_set_bit(&notif_bitmap, key);
+		sys_bitarray_set_bit(data->notif_bitmap, key);
 	}
 	k_spin_unlock(&data->notif_lock, sp_key);
 
@@ -473,7 +467,7 @@ static int optee_notif_wait(const struct device *dev, uint32_t key)
 	 * If notif bit was set then SEND command was already received.
 	 * Skipping wait.
 	 */
-	rc = sys_bitarray_test_and_clear_bit(&notif_bitmap, key, &prev_val);
+	rc = sys_bitarray_test_and_clear_bit(data->notif_bitmap, key, &prev_val);
 	if (rc || prev_val) {
 		goto out;
 	}
@@ -1164,12 +1158,22 @@ static const struct tee_driver_api optee_driver_api = {
 	.suppl_send = optee_suppl_send,
 };
 
+/*
+ * Bitmap of the ongoing notificatons, received from OP-TEE. Maximum number is
+ * CONFIG_OPTEE_MAX_NOTIF. This bitmap is needed to handle case when SEND command
+ * was received before WAIT command from OP-TEE. In this case WAIT will not create
+ * locks.
+ */
 #define OPTEE_DT_DEVICE_INIT(inst)					\
+	SYS_BITARRAY_DEFINE_STATIC(notif_bitmap_##inst, CONFIG_OPTEE_MAX_NOTIF); \
+									\
 	static struct optee_driver_config optee_config_##inst = {	\
 		.method = DT_INST_PROP(inst, method)			\
 	};								\
 									\
-	static struct optee_driver_data optee_data_##inst = {0};	\
+	static struct optee_driver_data optee_data_##inst = {		\
+		.notif_bitmap = &notif_bitmap_##inst			\
+	};								\
 									\
 	DEVICE_DT_INST_DEFINE(inst, optee_init, NULL, &optee_data_##inst, \
 			      &optee_config_##inst, POST_KERNEL,	\

--- a/drivers/tee/optee/optee.c
+++ b/drivers/tee/optee/optee.c
@@ -53,6 +53,10 @@ typedef void (*smc_call_t)(unsigned long a0, unsigned long a1, unsigned long a2,
 			   unsigned long a4, unsigned long a5, unsigned long a6, unsigned long a7,
 			   struct arm_smccc_res *res);
 
+struct optee_driver_config {
+	const char *method;
+};
+
 struct optee_notify {
 	sys_dnode_t node;
 	uint32_t key;
@@ -81,13 +85,13 @@ struct optee_supp {
 	struct k_sem reqs_c;
 };
 
-static struct optee_driver_data {
+struct optee_driver_data {
 	smc_call_t smc_call;
 
 	sys_dlist_t notif;
 	struct k_spinlock notif_lock;
 	struct optee_supp supp;
-} optee_data;
+};
 
 /* Wrapping functions so function pointer can be used */
 static void optee_smccc_smc(unsigned long a0, unsigned long a1, unsigned long a2, unsigned long a3,
@@ -1111,16 +1115,15 @@ static int optee_suppl_send(const struct device *dev, unsigned int ret, unsigned
 	return 0;
 }
 
-static int set_optee_method(void)
+static int set_optee_method(const struct device *dev)
 {
-	const char *method;
+	const struct optee_driver_config *conf = dev->config;
+	struct optee_driver_data *data = dev->data;
 
-	method = DT_PROP(DT_INST(0, DT_DRV_COMPAT), method);
-
-	if (!strcmp("hvc", method)) {
-		optee_data.smc_call = optee_smccc_hvc;
-	} else if (!strcmp("smc", method)) {
-		optee_data.smc_call = optee_smccc_smc;
+	if (!strcmp("hvc", conf->method)) {
+		data->smc_call = optee_smccc_hvc;
+	} else if (!strcmp("smc", conf->method)) {
+		data->smc_call = optee_smccc_smc;
 	} else {
 		LOG_ERR("Invalid smc_call method");
 		return -EINVAL;
@@ -1133,7 +1136,7 @@ static int optee_init(const struct device *dev)
 {
 	struct optee_driver_data *data = dev->data;
 
-	if (set_optee_method()) {
+	if (set_optee_method(dev)) {
 		return -ENOTSUP;
 	}
 	/*
@@ -1161,14 +1164,16 @@ static const struct tee_driver_api optee_driver_api = {
 	.suppl_send = optee_suppl_send,
 };
 
-/*
- * TODO This should be rewritten using DT_INST_FOREACH_STAUS_OKAY macro.
- * According to the OP-TEE documentation it is possible to have multiple
- * OP-TEE driver instances in the system. Right now only one instance is
- * supported, which may lead to possible issues in future.
- * For example notif_bitmap array should be moved from the global definition
- * to the DT_INST_FOREACH_STATUS_OKAY macro handler so each driver instance
- * will have it's own bitmap allocated.
- */
-DEVICE_DT_INST_DEFINE(0, optee_init, NULL, &optee_data, NULL, POST_KERNEL,
-		      CONFIG_KERNEL_INIT_PRIORITY_DEVICE, &optee_driver_api);
+#define OPTEE_DT_DEVICE_INIT(inst)					\
+	static struct optee_driver_config optee_config_##inst = {	\
+		.method = DT_INST_PROP(inst, method)			\
+	};								\
+									\
+	static struct optee_driver_data optee_data_##inst = {0};	\
+									\
+	DEVICE_DT_INST_DEFINE(inst, optee_init, NULL, &optee_data_##inst, \
+			      &optee_config_##inst, POST_KERNEL,	\
+			      CONFIG_KERNEL_INIT_PRIORITY_DEVICE,	\
+			      &optee_driver_api);			\
+
+DT_INST_FOREACH_STATUS_OKAY(OPTEE_DT_DEVICE_INIT)

--- a/tests/drivers/tee/optee/src/main.c
+++ b/tests/drivers/tee/optee/src/main.c
@@ -66,6 +66,13 @@ void arm_smccc_smc(unsigned long a0, unsigned long a1, unsigned long a2, unsigne
 	}
 }
 
+/* Allocate dummy arm_smccc_hvc function for the tests */
+void arm_smccc_hvc(unsigned long a0, unsigned long a1, unsigned long a2, unsigned long a3,
+		   unsigned long a4, unsigned long a5, unsigned long a6, unsigned long a7,
+		   struct arm_smccc_res *res)
+{
+}
+
 ZTEST(optee_test_suite, test_get_version)
 {
 	int ret;

